src/SFA.DAS.Commitments.Support.SubSite/Controllers/ApprenticeshipsController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+using System.Web.Mvc;
+using SFA.DAS.Commitments.Support.SubSite.Models;
+using SFA.DAS.Commitments.Support.SubSite.Orchestrators;
+using SFA.DAS.Commitments.Support.SubSite.Extensions;
+using SFA.DAS.Support.Shared;
+using SFA.DAS.Commitments.Support.SubSite.Enums;
+using Mvc = System.Web.Mvc;
+
+namespace SFA.DAS.Commitments.Support.SubSite.Controllers
+{
+    public class ApprenticeshipsController : Controller
+    {
+        private readonly IApprenticeshipsOrchestrator _orchestrator;
+
+        public ApprenticeshipsController(IApprenticeshipsOrchestrator apprenticeshipsOrchestrator)
+        {
+            _orchestrator = apprenticeshipsOrchestrator;
+        }
+
+        public async Task<ActionResult> Search(ApprenticeshipSearchQuery searchQuery)
+        {
+            if (string.IsNullOrWhiteSpace(searchQuery.SearchTerm))
+            {
+                return View(searchQuery);
+            }
+
+            switch (searchQuery.SearchType)
+            {
+                case ApprenticeshipSearchType.SearchByUln:
+                    return await UlnSearch(searchQuery);
+
+                case ApprenticeshipSearchType.SearchByCohort:
+                    return await CohortSearch(searchQuery);
+            }
+
+            return View(searchQuery);
+        }
+
+        [Mvc.Route("Apprenticeships/{Id}/account/{accountId}", Name = "ApprenticeshipDetails")]
```

Comment:
Would it be clearer to rename Id to HashedApprenticeshipId?

Reply:
To me, users of the system should not be aware of the fact that we are hashing the id. It is a system level thing. So if I send the Url to someone it should say Id.

Reply:
The user will see the actual hashed id in the url, not the parameter name. Renaming it will make it clearer to developers reading the code what it contains.

Reply:
Forgot I have changed it to use a custom route. Good point.